### PR TITLE
CUDA: Reduce memory pressure from local memory tests

### PR DESCRIPTION
**Pull request by gmarkall**
Local memory usage requires a reservation of (bytes per thread) * (number of SMs) * (Max resident threads per SM), which for kernels with local memory requirements of 80KB per thread (as is the case for these tests) forces allocations into the low GBs - this is a problem on the buildfarm as multiple tests run concurrently and can result in out-of-memory conditions.

This commit reduces the local memory allocations by a factor of 10 - I worry that reducing by another order of magnitude might cause local memory not to be allocated in some cases.

Some copy/paste errors that slipped through in these tests are also resolved (including the name of one of the tests!).

### CODE REVIEW

**Comment by gmarkall** [comment]
gpuci run tests

File: numba/cuda/tests/cudapy/test_dispatcher.py [hunk 1]
```diff
@@ -412,16 +412,16 @@ def simple_lmem(ary):
...
         # provides the same values as getting the shared mem per block for
         # individual signatures.
-        sh_mem_f32_all = simple_lmem.get_local_mem_per_thread()
-        sh_mem_f64_all = simple_lmem.get_local_mem_per_thread()
-        self.assertEqual(sh_mem_f32_all[sig_f32.args], local_mem_f32)
-        self.assertEqual(sh_mem_f64_all[sig_f64.args], local_mem_f64)
+        local_mem_f32_all = simple_lmem.get_local_mem_per_thread()
+        local_mem_f64_all = simple_lmem.get_local_mem_per_thread()
+        self.assertEqual(local_mem_f32_all[sig_f32.args], local_mem_f32)
+        self.assertEqual(local_mem_f64_all[sig_f64.args], local_mem_f64)
```
Comment (stuartarchibald) [inline numba/cuda/tests/cudapy/test_dispatcher.py:418]:
```suggestion
        local_mem_all = simple_lmem.get_local_mem_per_thread()
        self.assertEqual(local_mem_all[sig_f32.args], local_mem_f32)
        self.assertEqual(local_mem_all[sig_f64.args], local_mem_f64)
```
Is this just the same dictionary being assigned to different variables, in which case the above should be equivalent?

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the patch. One minor thing to resolve else looks good. The main test of this will be in CI to see if the resource pressure is reduced when the tests are run in a multi-process environment against the same device(s).

File: numba/cuda/tests/cudapy/test_dispatcher.py (hunk 1 above)
Comment (gmarkall) [reply numba/cuda/tests/cudapy/test_dispatcher.py:418]: Whoops, yes.

**Comment by gmarkall** [comment]
gpuci run tests

**Comment by gmarkall** [comment]
@stuartarchibald Many thanks for the review, suggestion committed.

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch and fixes.

**Comment by esc** [comment]
`Build numba_smoketest_cuda_yaml_170 has started`

**Comment by stuartarchibald** [comment]
> `Build numba_smoketest_cuda_yaml_170 has started`

Passed.

**Pull request merged**